### ISSUE TITLE
Fluent API does not use own Assertion class

### ISSUE THREAD

**Issue by dmolineus** [beberlei/assert#57]
There is no way to use a specific Assertion class when using the fluent api. So using fluent api andd an own excpetion type at the same time have a drawback to me. When changing the assertion a differnt exception will be thrown.

``` php
try {
    \My\Custom\Assertion::notNull(null);
    // throws \My\Custom\Assertion::$exceptionClass

    \Assert\that(null)->notNull();
    // throws \Assert\Assertion::$exceptionClass
}
catch(\My\Custom\AssertionFailedException $e) {
    // ...
}
```

**Comment by sagikazarmark** [comment]
I can imagine a solution like this:

``` php
\Assert\withCustom('CustomAssertionClass')
    ->that(null)
    ->notNull();
```

**Comment by bicpi** [comment]
+1 I'd also like to use my own exception class in a clean way when switching to the fluent interface.

**Comment by Solinx** [comment]
+1

**Comment by gries** [comment]
:+1:

**Comment by acgrid** [comment]
It headaches me too.
I forked it with my implementation here https://github.com/acgrid/assert which everything can be inherited and work with each other.
The significant issue to me is helper functions, I added two getter/setter functions to provide an interface to access which class to be used. 
To make it production usable, I made the fork namespaced (avoid possible collusion with another composer component) and tagged version x.1.
Since the change would be BC-breaking, so I am not confident to PR.
